### ISSUE TITLE
Respect `JsonProperty#required` when mapping inputs

### ISSUE THREAD

**Issue by kaqqao** [leangen/graphql-spqr#287]
Since Jackson meta-data is normally respected, `JsonProperty#required` should apply as well, and result in a non-null input field.

**Comment by retrodaredevil** [comment]
I disagree with this. Depending on what the the `JsonProperty` annotation is for, whoever uses this library may want to make sure that the JSON to deserialize contains that property, even if that property is null. Take this JSON for instance:

```
{
  "myValue": null
}
```
And this JsonCreator:
```
@JsonCreator
public MyValueClass(@JsonProperty(value = "myValue", required = true)Integer value){
  ...
}
```
It would not affect my code or anything, but I think just adding an extra annotation isn't too big a deal in this case.
